app/src/components/GenericModalComponent.tsx

```diff
@@ -44,46 +44,59 @@ const GenericModalComponent: React.FC<GenericModalProps> = ({
   if (!isVisible) return null;
 
   return (
-    <div className="modal-overlay">
-      <div className="modal-content">
-        {/* Title */}
-        <div className="modal-title">
-          {title !== null ? <h2>{title}</h2> : null}
+    <div className={`modal ${isVisible ? "modal-open" : null}`}>
+      <div className="modal-box bg-[#f1f9fb] max-h-80 hax-w-80 m-10 p-10">
+        {title !== null ? (
+          <div className="w-full mb-5">
+            <span className="font-bold text-xl text wrap">{title}</span>
+          </div>
+        ) : null}
 
-          {/* Window close button (X) */}
-          <button onClick={onClose} className="close-button">
-            &times;
-          </button>
-        </div>
+        {/* Window close button (X) */}
+        <button
+          onClick={onClose}
+          className="btn btn-sm btn-circle btn-ghost absolute right-4 top-4"
+        >
+          <MdClose size={25} className="text-slate-400" />
+        </button>
 
         {/* Content */}
-        <div className="modal-body">
-          {title !== null ? <p>{contentText}</p> : null}
-          {children}
-        </div>
+        {contentText !== null ? (
+          <p className="whitespace-normal w-full">{contentText}</p>
+        ) : null}
+        {children}
 
         {/* Action buttons */}
         {/* Only the cancel button is rendered (and centered), if text for confirmation button isn't defined */}
         <div
-          className={`action-buttons ${
+          className={`flex w-full mt-5 ${
             confirmBtnText === "" ? "justify-center" : "justify-between"
           }`}
         >
           {/* Cancel */}
-          <button type="button" className="cancel-button" onClick={onClose}>
-            {cancelBtnText}
+          <button
+            type="button"
+            className="flex flex-col w-40 btn bg-transparent border-none hover:bg-transparent px-0"
+            onClick={onClose}
+          >
+            <span className="text-[#166276] normal-case text-lg">
+              {cancelBtnText}
+            </span>
+            <hr className="w-3/5 border-[#166276]" />
```

Comment:
Remember to use the custom Tailwind CSS class 'bg-primary' here instead of the hex-color code ;)

Reply:
True! I didnt even think of that, I might just do a find and replace all on the entire repo tbh 😆